### PR TITLE
Add Sweep project tvl adapter

### PR DESCRIPTION
**Pull request by CelaPablo**
##### Name (to be shown on DefiLlama):
Sweep

##### Twitter Link:
https://twitter.com/SweeprFi

##### List of audit links if any:
https://github.com/SweeprFi/sweepr-contracts/tree/main/audits/sublime

##### Website Link:
https://sweepr.finance

##### Logo (High resolution, will be shown with rounded borders):
OP1: https://drive.google.com/file/d/1PjPSHYBy8CcanLwcMEobPgfbTIPmEJC5/view?usp=sharing
OP2: https://ibb.co/0t0c3pz

##### Current TVL:
23495

##### Treasury Addresses (if the protocol has treasury)
arb:0x7c9131d7E2bEdb29dA39503DD8Cf809739f047B3
eth:0x7c9131d7E2bEdb29dA39503DD8Cf809739f047B3
opt:0x7c9131d7E2bEdb29dA39503DD8Cf809739f047B3

##### Chain:
Arbitrum
Ethereum
Optimism

##### Short Description (to be shown on DefiLlama):
The Sweep liquidity layer eliminates cash drag and funds RWAs and L2s

##### Token address and ticker if any:
arb:0x89B1e7068bF8E3232dD8f16c35cAc45bDA584f4E, SWEEPR

##### Category (full list at https://defillama.com/categories) *Please choose only one:
RWA

##### forkedFrom (Does your project originate from another project):
NO

##### methodology (what is being counted as tvl, how is tvl being calculated):
Dollar assets converted to the SWEEP yield-bearing coin

##### Github org/user (Optional, if your code is open source, we can track activity):
SweeprFi - https://github.com/SweeprFi

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/sweep exports TVL: 
        
 
 
    optimism                  0
    arbitrum                  7.43 k
    ethereum                  7.54 k
    
    total                    14.96 k

**Comment by g1nt0ki** [comment]
sorry, we cant take usd value of sweep supply as tvl, where are the tokens backing sweep kept?

**Comment by CelaPablo** [comment]
The tokens backing SWEEP are in Stabilizer/Assets contracts.

We will fetch each Stabilizer's assetValue() that would bring only assets held in the contract that are not SWEEP, plus USDC holdings of each contract.

Will modify the script shortly. Thanks.

**Comment by llamatester** [comment]
The adapter at projects/sweep exports TVL: 
        
 
 
    ethereum                  12.03 k
    arbitrum                  8.04 k
    optimism                  3.40 k
    
    total                    23.47 k

File: projects/sweep/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      abi: 'function balanceOf(address) external view returns(uint256)',
+      calls: [{ target: USDC_ADDRESS, params: minter }]
+    });
+  });
+
+  const assetValueCalls = minters[0].map((minter) => {
+    return api.multiCall({ abi: 'uint256:assetValue', calls: [minter] });
+  });
+
+  const valueCalls = await Promise.all([...balanceOfCalls, ...assetValueCalls]);
```
Comment (g1nt0ki) [inline projects/sweep/index.js:33]: wait, wouldnt the `assetValueCalls` and `balanceOfCalls` have duplicated value? like same USDC is counted under both?

Comment (CelaPablo) [reply projects/sweep/index.js:33]: assetValueCalls only looks at holdings that are not USDC nor SWEEP. It represents the value of an investment in an external protocol, like USD-Plus tokens or sDAI tokens.
Code example here: https://github.com/SweeprFi/sweepr-contracts/blob/main/contracts/Assets/USDPlusAsset.sol#L62
Deployment of that here: https://arbiscan.io/address/0x52D0a9E74fC159F47cEE668801082a975c10bBba

**Pull request merged**